### DISCUSSION TITLE
`TypeError: expected string or bytes-like object` error when trying log model hyperparams. on mlflow

### DISCUSSION THREAD

**Discussion by alzaia** [mlflow/mlflow#8014] [Q&A]
Hello, I am having this error while logging my hyperparams. on my mlflow logger (using Pytorch Lightning). I have my Lightning model where I save my input params using:
`self.save_hyperparameters("in_channels", "out_channels")`

Printing the self.hparams of my model gives the expected content:
`"in_channels":      3` and  `"out_channels":      2`

Yet mlflow throws an error:
`python3.10/site-packages/mlflow/utils/validation.py", line 150, in _validate_param
    _validate_param_name(key)
  python3.10/site-packages/mlflow/utils/validation.py", line 217, in _validate_param_name
    if not _VALID_PARAM_AND_METRIC_NAMES.match(name):
TypeError: expected string or bytes-like object`

I have used this before on other projects and it has worked fine, am I missing something?

**Comment by JHibbard** [comment]
Hi alzaia,

What is returned when you run `type(self.hparams)`? It may be that the print method is doing some automatic type conversion that MLflow's logging function isn't.

Let me know if that helps and we can investigate further if not,
Jim

**Reply by alzaia** [reply]
Thanks for the reply @JHibbard ! The `self.hparams` return is a class of type `<class 'pytorch_lightning.utilities.parsing.AttributeDict'>`. The thing is, I have been using this way of logging hparams before in a pytorch lightning class (using the `MLFlowLogger` class from Lightning) and it has worked fine. Not sure what I'm doing wrong this time around. But I guess my error must be from the Lightning side rather than the MLFlow side.

**Comment by alzaia** [comment]
@JHibbard Interestingly when I just specify to log a `string` hparam, it does not throw an error but in my list of mlflow params it shows the value on both name and value fields. For instance, logging a string hparam:
`self.save_hyperparameters("activation")`
Then on the UI of mlflow:
Name | Value
-- | --
ReLU | ReLU
